unified_planning/engines/factory.py

```diff
@@ -20,18 +20,22 @@
 import unified_planning as up
 from unified_planning.environment import Environment, get_env
 from unified_planning.model import ProblemKind
+from unified_planning.engines.mixins.oneshot_planner import OptimalityGuarantee
+from unified_planning.engines.mixins.compiler import CompilationKind
 from typing import IO, Dict, Tuple, Optional, List, Union, Type
 
 
-DEFAULT_SOLVERS = {'enhsp' : ('up_enhsp', 'ENHSPsolver'),
-                   'fast_downward' : ('up_fast_downward', 'FastDownwardPDDLSolver'),
-                   'fast_downward_optimal' : ('up_fast_downward', 'FastDownwardOptimalPDDLSolver'),
-                   'lpg' : ('up_lpg', 'LPGsolver'),
-                   'tamer' : ('up_tamer.solver', 'SolverImpl'),
-                   'pyperplan' : ('up_pyperplan.solver', 'SolverImpl'),
-                   'sequential_plan_validator' : ('unified_planning.solvers.plan_validator', 'SequentialPlanValidator'),
-                   'up_grounder' : ('unified_planning.solvers.grounder', 'Grounder'),
-                   'tarski_grounder' : ('unified_planning.solvers.tarski_grounder', 'TarskiGrounder')}
+DEFAULT_ENGINES = {
+    'pyperplan' : ('up_pyperplan.engine', 'EngineImpl'),
+    'tamer' : ('up_tamer.engine', 'EngineImpl'),
+    'sequential_plan_validator' : ('unified_planning.engines.plan_validator', 'SequentialPlanValidator'),
+    'up_conditional_effects_remover' : ('unified_planning.engines.compilers.conditional_effects_remover', 'ConditionalEffectsRemover'),
```

Comment:
Since we now support more than one compilation kind, can we have an engine called `embedded` that offers all the available `up_**` and `sequential_plan_validator` compilation kinds?